### PR TITLE
feat: add `--color-scale`

### PR DESCRIPTION
**Pull request by cafkafk**
##### Description

This is based on a series of commits done by aashirabadb in #526.

This introduces the `--color-scale` flag both for file age and size, as
well as `--color-scale-mode` which allows "fixed" and gradient".

The reason for reapplying from the diff of the PR instead of the patch
is that the commit history was messy, and introduced "breaking changes"
that were only locally breaking to the PR, that is, not the users.

Further, the PR had seemingly gone stale, and the rather long and
complicated commit history interwoven with merges made it seem more
efficient to just work from scratch.

Again, the work here is done by aashirbadb, my contribution is just
ensuring the quality of the code they've written.

Co-authored-by: aashirbadb <aashirbadbhandari@gmail.com>
Signed-off-by: Christina Sørensen <christina@cafkafk.com>

Refs: #486

Closes: https://github.com/eza-community/eza/issues/486

##### How Has This Been Tested?

Integration tests via `just itest`. Generation of man pages has also been tested with `just man`, and inspected, and seem fine. Testing of introduced flags was also performed, as well as basic functionality not having obvious regressions. Basic testing of performance regressions was also performed, and nothing indicated regressions.

Will be further investigated before merge.

### CODE REVIEW

**Comment by cohml** [comment]
Great work guys. Glad this got implemented so quickly. I look forward to the release.

File: man/eza.1.md [hunk 1]
```diff
@@ -105,10 +112,15 @@ The default value is ‘`automatic`’.
 `-w`, `--width=COLS`
 : Set screen width in columns.
 
+Valid options are `none`, `absolute` or `relative`.
+The default value is `none`
+
+`absolute` mode highlights based on file modification time relative to the past year.
+`relative` mode highlights based on file modification time in relation to other files. `none` disables highlighting.
```
Comment (bew) [inline man/eza.1.md:119]: I don't understand this added hunk, what option does it document? @cafkafk

File: man/eza.1.md [hunk 2]
```diff
@@ -302,6 +314,9 @@ For more information on the format of these environment variables, see the [eza_
 
 Overrides any `--git` or `--git-repos` argument
 
+## `EZA_MIN_LUMINANCE`
+Specifies the minimum luminance to use when decay is active. It's value can be between -100 to 100.
```
Comment (bew) [inline man/eza.1.md:318]: I think it would be useful to have an example of use of this, qnd whqr is means to set values like -100, -50, 0, 23, 100

**Comment by cafkafk** [comment]
This seem to have obvious improvement opportunities. Perhaps we should work on those in a new issue/pr

**Comment by eggbean** [comment]
Would you consider making an option to use the previous system? I'm finding this quite distracting and my directories usually have most file sizes barely readable and then a couple being very bright and visible. Would it be better if the default colours were much more similar rather than so contrasting?

**Comment by cohml** [comment]
The contrast is kind of the point though, no?

If it were to be decreased, I feel like that should be configurable and not a fixed default.

**Comment by eggbean** [comment]
Why should contrast be the point? It makes much more sense for the differences to be subtle (as before) - that's all that's needed to indicate the information being expressed.

I'm wondering if what I am experiencing is what is intended, as I cannot understand why anyone would think this is good? This is what it looks like (unsetting my `$EXA_COLORS` changes):

image

**Comment by cohml** [comment]
Perhaps it's possible to expose another argument, `--color-scale-contrast` or something, by which you could control it.

Say, passing `10` would result in the current full contrast, with the contrast decreasing with smaller numbers, and `0` being equivalent to the previous (pre-decay) behavior?

This would be nicer with 0-1 instead of 0-10, but I know floating point values can be tricky with the CLI.

**Comment by eggbean** [comment]
That sounds like feature creep/bloat to me. Something like this doesn't need to be so complicated. It's better is someone just makes a good design decision that will suit most users, in my opinion. Anyway, it looks like I will just stick with v0.15.2 unless this is improved.

**Comment by cohml** [comment]
> That sounds like feature creep/bloat to me.

TBH I definitely agree with that.

The current (v0.15.3) set of flags allows for this entire feature to be turned off anyway, if I'm not mistaken.

**Comment by cafkafk** [comment]
Idk I'm not a big fan of the colors either I just don't dare promise to improve them right now my todo list is already very long

**Pull request merged**